### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by HannaHurieva**
1. Сhanged users entity (from Liza's version), created enum Role Based Access Control, added table "user_role" in database
2. added implementation UserDetails for class Users, created class UserDetailsServiceImpl implements UserDetailsService, added implementation GrantedAuthority for enum Role
3. Connected user and spring security : UserDetailService injected in to WebSecurityConfig along with the passsword encoding using BCrypt
4. Created enum Status, updated class User, changed columns settings
5. Added the ability to register a user with recording data in the DB, created Registration form with help Freemarker Template Engine, created Registration controller saves users with unique username and email

### CODE REVIEW

File: src/main/java/com/alevel/project/coffee/service/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+
+import static com.alevel.project.coffee.model.enums.Status.ACTIVE;
+
+@Service
+public class UserServiceImpl {
```
Comment (ybotok) [inline src/main/java/com/alevel/project/coffee/service/UserServiceImpl.java:15]: what implement?

File: src/main/resources/templates/login.ftl [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<#import "parts/common.ftl" as c>
+<#import "parts/login.ftl" as l>
+
+<@c.page>
+    Login page
+    <@l.login "/login" />
+    <a href="/registration">Add new user</a>
+</@c.page>
```
Comment (ybotok) [inline src/main/resources/templates/login.ftl:8]: plz, check new-line

File: src/main/java/com/alevel/project/coffee/service/UserServiceImpl.java (hunk 1 above)
Comment (HannaHurieva) [reply src/main/java/com/alevel/project/coffee/service/UserServiceImpl.java:15]: at the branch re-captcha was added UserService interface and updated service layer

File: src/main/resources/templates/login.ftl (hunk 2 above)
Comment (HannaHurieva) [reply src/main/resources/templates/login.ftl:8]: ok, at the branch registrationTest

**Pull request merged**